### ISSUE TITLE
cargo install of v0.8.1-bis branch fails

### ISSUE THREAD

**Issue by ximon18** [NLnetLabs/krill#391]
Tested using Docker CentOS 7 and Ubuntu 20.04 with latest Rust 1.49.0.

Happens both with default featues and `--features multi-user`.

# CentOS 7

```
docker run -it centos:7

# yum -y update
# yum group install "Development Tools"
# yum install openssl-devel
# curl https://sh.rustup.rs -sSf | sh
# source $HOME/.cargo/env
# rustc --version
rustc 1.49.0 (e1884a8e3 2020-12-29)
# cargo install --git https://github.com/NLnetLabs/krill --branch v0.8.1-bis
    Updating git repository `https://github.com/NLnetLabs/krill`
  Installing krill v0.8.1-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#1d1935e6)
    Updating crates.io index
   Compiling libc v0.2.82
...
error[E0603]: module `export` is private
   --> src/commons/remote/rfc6492.rs:7:12
    |
7   | use serde::export::fmt::Display;
    |            ^^^^^^ private module
    |
note: the module `export` is defined here
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.119/src/lib.rs:275:5
    |
275 | use self::__private as export;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0603`.
error: could not compile `krill`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `krill v0.8.1-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#1d1935e6)`, intermediate artifacts can be found at `/tmp/cargo-installg8GPce`

Caused by:
  build failed
```

# Ubuntu 20.04

```
docker run -it ubuntu:20.04
# apt update -y
# apt upgrade -y
# apt install -y curl build-essential libssl-dev openssl pkg-config
# curl https://sh.rustup.rs -sSf | sh
# source $HOME/.cargo/env
# rustc --version
rustc 1.49.0 (e1884a8e3 2020-12-29)
# cargo install --git https://github.com/NLnetLabs/krill --branch v0.8.1-bis --features multi-user
    Updating git repository `https://github.com/NLnetLabs/krill`
  Installing krill v0.8.1-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#1d1935e6)
    Updating crates.io index
  Downloaded tinyvec_macros v0.1.0
  Downloaded tinyvec v1.1.0
  Downloaded unicode-normalization v0.1.16
  Downloaded 3 crates (133.7 KB) in 0.38s
   Compiling libc v0.2.82
...
   Compiling oso v0.8.2
error[E0603]: module `export` is private
   --> src/commons/remote/rfc6492.rs:7:12
    |
7   | use serde::export::fmt::Display;
    |            ^^^^^^ private module
    |
note: the module `export` is defined here
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.119/src/lib.rs:275:5
    |
275 | use self::__private as export;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0603`.
error: could not compile `krill`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `krill v0.8.1-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#1d1935e6)`, intermediate artifacts can be found at `/tmp/cargo-installuUaTNI`

Caused by:
  build failed 
```

**Comment by ximon18** [comment]
Note: passing `--locked` to `cargo install` "solves the problem:

On Ubuntu 20.04:

```
# cargo install --git https://github.com/NLnetLabs/krill --branch v0.8.1-bis --locked
    Updating git repository `https://github.com/NLnetLabs/krill`
  Installing krill v0.8.1-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#1d1935e6)
    Updating crates.io index
warning: package `miow v0.2.1` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
warning: package `net2 v0.2.35` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
warning: package `socket2 v0.3.15` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
  Downloaded cc v1.0.61
...
   Installed package `krill v0.8.1-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#1d1935e6)` (executables `krill`, `krillc`, `krillpubc`, `krillpubd`)
```

**Comment by ximon18** [comment]
The issue can be reproduced without `cargo install` like so:

```
$ cargo update -p serde
$ cargo build
```

The problem appears to be this line from [rrc6492.rs](https://github.com/NLnetLabs/krill/blob/v0.8.1-bis/src/commons/remote/rfc6492.rs#L7) in Krill:

```
use serde::export::fmt::Display;
```

This import works with serde 1.0.117 (specified in Krill `Cargo.lock`) but not with the newest version 1.0.119 (which `cargo install` selects as it matches the version filter in `Cargo.toml` for serde which says `version = "^1.0"`).

Replacing the import with a direct dependency on `std::fmt::Display` seems to solve the problem.

**Comment by ximon18** [comment]
Verified that with the fix merged to the v0.8.1-bis branch that the failure no longer occurs:

```
docker run -it centos:7

# yum -y update
# yum group install "Development Tools"
# yum install openssl-devel
# curl https://sh.rustup.rs -sSf | sh
# source $HOME/.cargo/env
# rustc --version
rustc 1.49.0 (e1884a8e3 2020-12-29)
# cargo install --git https://github.com/NLnetLabs/krill --branch v0.8.1-bis
    Updating git repository `https://github.com/NLnetLabs/krill`
  Installing krill v0.8.2-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#c617e3ac)
    Updating crates.io index
  Downloaded cc v1.0.66
...
   Compiling reqwest v0.10.10
    Finished release [optimized] target(s) in 5m 28s
  Installing /root/.cargo/bin/krill
  Installing /root/.cargo/bin/krillc
  Installing /root/.cargo/bin/krillpubc
  Installing /root/.cargo/bin/krillpubd
   Installed package `krill v0.8.2-bis (https://github.com/NLnetLabs/krill?branch=v0.8.1-bis#c617e3ac)` (executables `krill`, `krillc`, `krillpubc`, `krillpubd`)
```
